pkg/test/framework/api/components/apps.go

```diff
@@ -43,35 +43,32 @@ type Apps interface {
 	GetAppOrFail(name string, t testing.TB) App
 }
 
+// Service represents a deployed service within the mesh.
+type Service interface {
+	ClusterIP() string
+}
+
 // App represents a deployed fake App within the mesh.
 type App interface {
 	Name() string
 	Endpoints() []AppEndpoint
 	EndpointsForProtocol(protocol model.Protocol) []AppEndpoint
 	Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedResponse, error)
 	CallOrFail(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse
+	Service() Service
+	CallURL(url *url.URL, dst App, opts AppCallOptions) ([]*echo.ParsedResponse, error)
```

Comment:
The problem with these APIs is that they won't work for both native and k8s tests. That's why we use the abstractions in Call/CallOrFail.

I think what you really need here is to augment the AppCallOptions and ultimately the app itself to support what you need.

Reply:
We removed the additional methods CallURL and CallURLOrFail